Update hal_nordic repository to its latest version

czeslawmakarski opened the pull request:
This PR updates hal_nordic repository to its latest version.

AdamZelikNS committed:
modules: nrfx_glue: Reserve PPI/DPPI resources used by 802.15.4 drv

Mark the PPI/DPPI channels and groups used by the 802.15.4 driver
as occupied and thus unavailable for allocation through nrfx_ppi.

Signed-off-by: Adam Zelik <adam.zelik@nordicsemi.no>

pkwiek committed:
modules: Change serialized radio init priority

This commit changes serialized radio initialization
priority. This fixes race condition where serialization
backend would boot earlier than the radio driver itself.

Signed-off-by: Pawel Kwiek <pawel.kwiek@nordicsemi.no>

pkwiek committed:
drivers: ieee802154: Enable CSMA-CA on serialized host

This commit enables CSMA-CA on the serialized host.

Signed-off-by: Pawel Kwiek <pawel.kwiek@nordicsemi.no>

czeslawmakarski committed:
west.yml: Update hal_nordic revision

This commit updates hal_nordic revision.

Signed-off-by: Czeslaw Makarski <Czeslaw.Makarski@nordicsemi.no>

czeslawmakarski committed:
drivers: ieee802154: Remove obsolete Kconfig entry.

This commit removes obsolete Kconfig entry.

Signed-off-by: Czeslaw Makarski <Czeslaw.Makarski@nordicsemi.no>

czeslawmakarski committed:
drivers: ieee802154: Make HW Radio Capabilities runtime

This commit introduces runtime query of the HW Capabilities of
the nRF IEEE 802.15.4 Radio Driver.

Signed-off-by: Czeslaw Makarski <Czeslaw.Makarski@nordicsemi.no>

czeslawmakarski commented:
Bug was discovered in this PR so merge is postponed until fix is decided.

czeslawmakarski committed:
samples: 802154_rpmsg: Remove redundant kconfig

This commit removes the obsolete hardcoded kconfig in the
802154_rpmsg sample.

Signed-off-by: Czeslaw Makarski <Czeslaw.Makarski@nordicsemi.no>

Commit:
west.yml: Update hal_nordic revision

This commit brings in several latest bugfixes for the hal_nordic repo.
It also removes a Kconfig entry that these changes yield deprecated.

Signed-off-by: Jedrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

czeslawmakarski committed:
samples: 802154_rpmsg: Add SL fault handler

This commit adds the nRF IEEE 802.15.4 Service Layer Fault Handler
to the 802154_rpmsg sample.

Signed-off-by: Czeslaw Makarski <Czeslaw.Makarski@nordicsemi.no>

hubertmis commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -390,7 +394,8 @@ static void nrf5_tx_started(const struct device *dev,
 
-#ifdef CONFIG_IEEE802154_NRF5_PKT_TXTIME
+/* This function cannot be used in the serialized version yet. */
+#if CONFIG_NET_PKT_TXTIME && !CONFIG_NRF_802154_SER_HOST
```
with booelan CONFIG entries it should be `#if defined(CONFIG_...) && !defined(CONFIG_...)`

jciupis replied:
Done.

Commit:
samples: boards: Add tests section to 802154_rpsmg sample

This commit adds tests section to sample.yaml file of the
ieee802154/802154_rpmsg sample.

Signed-off-by: Jedrzej Ciupis <jedrzej.ciupis@nordicsemi.no>
